app/controllers/catalog_controller.rb

```diff
@@ -355,18 +355,24 @@ def lazy_wps_workflow_facet
 
   def show
     # If showing a user version, druid will be :item_id and user_version will be :id.
-    @druid = Druid.new(params[:item_id] || params[:id]).with_namespace
-    @user_version = params.key?(:item_id) ? params[:id] : nil
-    _deprecated_response, @document = search_service.fetch(@druid)
-
-    @cocina = Repository.find_lite(@druid, structural: false)
+    if params.key?(:item_id)
+      @druid = Druid.new(params[:item_id]).with_namespace
+      @user_version = params[:id]
+      @cocina = Repository.find_user_version(@druid, @user_version)
+      @document = SolrDocument.new(object_client.user_version.solr(@user_version))
```

Comment:
DSA is creating the solr document on the fly. It is not added to the solr index.